internal/parse/parts.go

```diff
@@ -0,0 +1,68 @@
+package parse
+
+// A queryPart represents a section of a parsed SQL statement, which forms
+// a complete query when processed together with its surrounding parts, in
+// their correct order.
+type queryPart interface {
+	// String returns the part's representation for debugging purposes.
+	String() string
+
+	// ToSQL returns the SQL representation of the part.
+	ToSQL() string
+}
+
+// FullName represents a table column or a Go type identifier.
+type FullName struct {
+	Prefix, Name string
+}
+
+// InputPart represents a named parameter that will be send to the database
+// while performing the query.
+type InputPart struct {
+	FullName
```

Comment:
I'm lacking a bit of context on usage, but gut feeling is that I'd make this symmetrical with OutputPart, having a field name here instead of embedding. Maybe Source?

Reply:
Done. `Source` and `Target` make more sense for the `OutputPart` but it is not confusing here either, so I suppose we can use it for symmetry.